compiler/rustc_codegen_ssa/src/mir/operand.rs

```diff
@@ -118,22 +118,20 @@ impl<'a, 'tcx, V: CodegenObject> OperandRef<'tcx, V> {
     }
 
     pub fn deref<Cx: LayoutTypeMethods<'tcx>>(self, cx: &Cx) -> PlaceRef<'tcx, V> {
+        if self.layout.ty.is_box() {
```

Comment:
Could you add a similar assertion to Miri?

Reply:
Also, does this mean you can remove the special cases I listed at https://github.com/rust-lang/rust/issues/95453#issuecomment-1101897141 ?

Reply:
The first 3 special cases have been removed. The debuginfo one probably shouldn't be changed because it could make debugging more annoying for no benefit. I'm currently investigating to see if the last one can be changed safely without breaking things.

Reply:
Also please remember to add a similar assertion to Miri. :)

Sorry for raising two points in the same thread. Sadly GitHub does not let one open multiple threads at the same line of code.